.circleci/config.yml

```diff
@@ -0,0 +1,150 @@
+version: 2
+
+references:
+  java_node_image: &java_node_image
+    docker:
+    - image: circleci/openjdk:8u181-jdk-node-browsers
+
+  attach_workspace: &attach_workspace
+    attach_workspace:
+      at: .
+
+  restore_cache: &restore_cache
+    restore_cache:
+      keys:
+      # when lock file changes, use increasingly general patterns to restore cache
+      - gradle-repo-v1-{{ .Branch }}-{{ checksum "build.gradle" }}
+      - gradle-repo-v1-{{ .Branch }}-
+      - gradle-repo-v1-
+
+  cf_install: &cf_install
+    run:
+      command: |
+        curl -L "https://packages.cloudfoundry.org/stable?release=linux64-binary&source=github" | tar -zx
+        mkdir -p `pwd`/bin
+        chmod +x ./cf
+        mv ./cf bin/
+
+  cf_login: &cf_login
+    run:
+      name: Login to Cloud Foundry
+      command: ./bin/cf login -o dta -s notifications -a https://api.system.y.cld.gov.au -u $CF_Y_USER -p $CF_Y_PASSWORD
```

Comment:
@andrewoh531 might be worth pulling out the `dta` org and `notifications` space into `CF_Y_ORG|SPACE` env vars? Just thinking that since people might fork these examples it's probably better those are not hard-coded. Probably worth pulling out `https://api.system.y.cld.gov.au` into `CF_ENDPOINT` too.